### ISSUE TITLE
Retry

### ISSUE THREAD

**Issue by nicholas22** [peichhorn/lombok-pg#3]
I have another suggestion, how about a "retry" construct, only callable from a catch block, which would allow a try block to be repeated, as you may have provided some kind of solution within the catch block. Perhaps an overload with the number of retries would also be beneficial, when e.g. attempting to do some networking task, such as connecting to a database. 

The implementation could be done using a goto jump in bytecode, or if you're attempting a set number of times, perhaps a for loop would be more appropriate...

**Comment by peichhorn** [comment]
Until now I never had to do this, but a simple loop should be sufficient for this, or am I missing something:

boolean retry = true;
while(retry) {
  retry = false;
  try {
    ...
  } catch (...) {
    ...
    retry = true;
  }
}

**Comment by nicholas22** [comment]
Yes, it's not very hard to do manually as you showed, but it does look somewhat ugly, especially if you're retrying for a set number of times, as you need a counter as well. It's less about reducing boilerplate, rather it introduces a pattern of how to do things. It's a concise 'nice-to-have', not a major feature, along the lines of e.g. the tuple feature.
